rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/terminal/LSPTerminalREPL.java

```diff
@@ -0,0 +1,129 @@
+        super(makeInterpreter(terminal, services), null, stdin, stderr, stdout, true, terminal.isAnsiSupported(), getHistoryFile(), terminal, null);
+    }
+
+    private static ILanguageProtocol makeInterpreter(Terminal terminal, final IDEServices services) throws IOException, URISyntaxException {
+        
+        RascalInterpreterREPL repl =
+            new RascalInterpreterREPL(prettyPrompt, allowColors, getHistoryFile()) {
+                @Override
+                protected Evaluator constructEvaluator(InputStream input, OutputStream stdout, OutputStream stderr) {
+                    Evaluator eval =  ShellEvaluatorFactory.getDefaultEvaluator(input, stdout, stderr);
+
+                    URIResolverRegistry reg = URIResolverRegistry.getInstance();
+
+                    reg.registerLogical(new ILogicalSourceLocationResolver(){
+                        @Override
+                        public ISourceLocation resolve(ISourceLocation input) throws IOException {
+                            return services.resolveProjectLocation(input);
+                        }
+
+                        @Override
+                        public String scheme() {
+                            return "project";
+                        }
+
+                        @Override
+                        public String authority() {
+                            return null;
+                        }
+                    });
+                    
+                    return eval;
+                }
+
+                @Override
+                public void handleInput(String line, Map<String, InputStream> output, Map<String, String> metadata)
+                    throws InterruptedException {
+                    super.handleInput(line, output, metadata);
+
+                    for (String mimetype : output.keySet()) {
+                        if (!mimetype.contains("html") && !mimetype.startsWith("image/")) {
+                            continue;
+                        }
+
+                        services.browse(URIUtil.assumeCorrect(metadata.get("url")));
+                    }
+                }
+            };
+    
+        repl.setMeasureCommandTime(false);
+    
+        return repl;
+    }
+
+    private static File getHistoryFile() throws IOException {
+        File home = new File(System.getProperty("user.home"));
+        File rascal = new File(home, ".rascal");
+    
+        if (!rascal.exists()) {
+            rascal.mkdirs();
+        }
+    
+        File historyFile = new File(rascal, ".repl-history-rascal-terminal");
+        if (!historyFile.exists()) {
+            historyFile.createNewFile();
+        }
+    
+        return historyFile;
+    }
+
+    public static void main(String[] args) throws InterruptedException {
+        int ideServicesPort = -1;
+
+        for (int i = 0; i < args.length; i++) {
+            if ("--ideServicesPort".equals(args[i])) {
+                ideServicesPort = Integer.parseInt(args[++i]);
```

Comment:
crashes if only `--ideServicesPort` is passed as argument (and no number afterwards).

Also we have to catch `NumberFormatException` in case of invalid formatting.